### PR TITLE
cilium: prandom based backend selection

### PR DESCRIPTION
**Pull request by borkmann**
/cc @brb ptal

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9111)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

File: bpf/lib/common.h [hunk 1]
```diff
@@ -560,7 +560,7 @@ struct lb6_service_v2 {
 	__u32 backend_id;
 	__u16 count;
 	__u16 rev_nat_index;
-	__u16 weight;
+	__u16 weight;		/* Currently unused. */
```
Comment (brb) [inline bpf/lib/common.h]: Nit: same comment for `lb4_service_v2`?

**Comment by borkmann** [comment]
test-me-please

**Review by brb — APPROVED** [review]
LGTM! Just one nit.

**Review by tgraf — APPROVED** [review]
LGTM

**Comment by borkmann** [comment]
test-me-please

File: bpf/lib/common.h (hunk 1 above)
Comment (borkmann) [reply bpf/lib/common.h]: We have all the comments in `lb4_service_v2`, so I removed the above snippet.

**Pull request merged**